buildmodel/buildmodel.go

```diff
@@ -275,3 +321,25 @@ func makeOSArchPlatform(os, osVersion string, env *dockerversions.ArchEnv) *dock
 		OSVersion:    osVersion,
 	}
 }
+
+// joinTag joins the given strings with "-" to form a Docker tag (or partial tag). Empty strings are
+// ignored and do not result in extra "-" characters. This is especially useful when it would be
+// inconvenient for the caller to keep track of which elements might be an empty string.
+func joinTag(s ...string) string {
+	if len(s) == 0 {
+		return ""
+	}
+	b := strings.Builder{}
```

Comment:
nit: normal variable declaration is preferred over shorthand syntax when declaring a variable with default initialization.

```suggestion
	var b strings.Builder
```

Reply:
Good catch. I think what's happening is that I type `x := `, look in the package for a `NewX` func if I haven't used `X` in a while, end up seeing that `X` is expected to be created directly, and keep moving forward with `{}` rather than revising the start of the line. 😄 Hopefully now I can keep an eye out for it better.

Reply:
This exact same usability issue happens to my every now and then, but once my eyes see this pattern, they command the fingers without going through the brain 😄